### ISSUE TITLE
Can I build DCNV2 on Windows? 

### ISSUE THREAD

**Issue by shoutOutYangJie** [CharlesShang/DCNv2#62]
I tried it, but failed. Is it possible if I run DCNV2 on windows?

**Comment by AndSonder** [comment]
I think it is impossible because the project will use .so file.

**Comment by seth-cohen** [comment]
Yes, it is possible I was just able to build. The process was a pain... I hope I capture it all.
I am running Windows 10 and used Visual Studio 2017 to build. 

- You want pytorch==1.4 and torchvision==0.5
- From within Visual Studio Installer click modify and install the following.
  - Individual Component > Windows Universal CRT SDK
  - Desktop Development with C++ 
  - As part of the Desktop Development with C++ you'll need the following optional components
    - Windows 10 SDK
    - VC++ 2017 toolset
- I had to update this line of code https://github.com/CharlesShang/DCNv2/issues/14. You may not have to depending on the version you are on.
- Add the following to your environment vars (your paths may be slightly different so pay special attention here).
  - INCLUDE="C:\Program Files (x86)\Windows Kits\10\Include\10.0.16299.0\ucrt;C:\Program Files (x86)\Windows Kits\10\Include\10.0.16299.0\shared"
  - LIB="C:\Program Files (x86)\Windows Kits\10\Lib\10.0.16299.0\um\x64;C:\Program Files (x86)\Windows Kits\10\Lib\10.0.16299.0\ucrt\x64"
  - Add this to your PATH "C:\Program Files (x86)\Windows Kits\10\bin\10.0.16299.0\x64"

After all of that I was able to get it to build by opening up `x64 Native Tools Command Prompt for VS 2017` navigating to the repo, activating my conda environment and running `python setup.py build develop`

**Comment by luyunxi** [comment]
@seth-cohen Hi, how i add INCLUDE、LIB and PATH on my window PC. THX!

**Comment by luyunxi** [comment]
@seth-cohen  Now, I got the issue:

corecrt.h”: No such file or directory
error: command 'D:\\software\\VS2017\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2

Do you have any solution?

**Comment by CandyCook** [comment]
> @ seth-cohen 现在，我得到了问题：
> 
> corecrt.h”：没有此类文件或目录
> 错误：命令“ D：\ software \ VS2017 \ VC \ Tools \ MSVC \ 14.16.27023 \ bin \ HostX86 \ x64 \ cl.exe”失败，退出状态为2
> 
> 你有什么解决办法吗？

I still have the same question. How can you solve it?

**Comment by rathaROG** [comment]
For all latest pytorchs with cuda on **Windows**:
https://github.com/rathaROG/DCNv2_Windows
